lib/chewy/type/adapter/active_record.rb

```diff
@@ -22,31 +22,65 @@ def cleanup_default_scope!
           @default_scope = @default_scope.reorder(nil).limit(nil).offset(nil)
         end
 
-        def import_scope(scope, batch_size)
-          scope = scope.reorder(target_id.asc).limit(batch_size)
+        def import_scope(scope, batch_size, sort_by_updated_at, &block)
+          if sort_by_updated_at && timestamp_column_name
+            import_scope_ordered_by_timestamp(scope, batch_size, &block)
+          else
+            import_scope_ordered_by_id(scope, batch_size, &block)
+          end
+        end
 
-          ids = pluck_ids(scope)
+        def import_scope_ordered_by_id(scope, batch_size)
           result = true
 
+          scope = scope.reorder(id_column.asc).limit(batch_size)
+
+          ids = pluck_ids(scope)
+
           while ids.present?
             result &= yield grouped_objects(default_scope_where_ids_in(ids))
             break if ids.size < batch_size
-            ids = pluck_ids(scope.where(target_id.gt(ids.last)))
+            ids = pluck_ids(scope.where(id_column.gt(ids.last)))
           end
 
           result
```

Comment:
After thinking about it for a while, I found a solution which might help to make the code a bit simpler and satisfy :cop:.

Didn't test it:

```ruby
def import_scope_ordered_by_timestamp(scope, batch_size)
  result = true

  # TODO: can the last 2 parameters be moved to TimestampOrderedBatches?
  TimestampOrderedBatches
    .new(scope, batch_size, id_column, timestamp_column)
    .each do |batch|
    result &= batch
  end

  result
end

class TimestampOrderedBatches
  def initialize(scope, batch_size, id_column, timestamp_column)
    @scope = scope
    @batch_size = batch_size
    @id_column = id_column
    @timestamp_column = timestamp_column
  end

  def each
    ids_and_dates = first_batch_ids_and_dates

    while ids_and_dates.present?
      yield grouped_objects(scope_where_ids_in(ordered_scope, ids_and_dates.map(&:first)))
      break if ids_and_dates.size < batch_size
      last_id, last_updated_at = ids_and_dates.last
      ids_and_dates = next_batch_ids_and_dates(last_id, last_updated_at)
    end
  end

  private

  attr_reader :scope, :batch_size, :id_column, :timestamp_column

  def first_batch_ids_and_dates
    pluck_ids_and_dates(first_batch_scope)
  end

  def next_batch_ids_and_dates(last_id, last_updated_at)
    pluck_ids_and_dates(next_batch_scope(last_id, last_updated_at))
  end

  def first_batch_scope
    ordered_scope
  end

  def next_batch_scope(last_id, last_updated_at)
    ordered_scope.where(next_timestamp_ordered_batch_condition(last_id, last_updated_at))
  end

  def pluck_ids_and_dates(scope)
    scope.except(:includes).uniq.pluck(id_column, timestamp_column)
  end

  def ordered_scope
    @ordered_scope ||= scope.reorder(timestamp_column_asc, id_column_asc).limit(batch_size)
  end
end
```

>